js/start.js

```diff
@@ -1,18 +1,60 @@
 import $ from 'jquery';
 import Backbone from 'backbone';
+import Polyglot from 'node-polyglot';
 import './lib/whenAll.jquery';
 import app from './app';
 import LocalSettings from './models/LocalSettings';
 import ObRouter from './router';
 import PageNav from './views/PageNav.js';
 import LoadingModal from './views/modals/Loading';
 import Dialog from './views/modals/Dialog';
+import { getLangByCode } from './data/languages';
 import Profile from './models/Profile';
 import Settings from './models/Settings';
 
+let beyondOnboarding = false;
+
 app.localSettings = new LocalSettings({ id: 1 });
 app.localSettings.fetch().fail(() => app.localSettings.save());
 
+// initialize language functionality
+function getValidLanguage(lang) {
+  if (getLangByCode(lang)) {
+    return lang;
+  }
+
+  return 'en-US';
+}
+
+const initialLang = getValidLanguage(app.localSettings.get('language'));
+app.localSettings.set('language', initialLang, { supressLangChangeWarning: true });
+app.polyglot = new Polyglot();
+app.polyglot.extend(require(`./languages/${initialLang}.json`));
+
+app.localSettings.on('change:language', () => {
```

Comment:
So the idea is that if we are beyond onboarding, any language changes will kick up a dialog telling the user they need to re-start before the changes will fully take effect. This allows us to simplify things where almost all of our views don't need to do anything to update themselves with a new lang (which in some cases was tricky).

If we're not beyond onboarding, then we will re-render the page-nav (so we need to be mindful to ensure page-nav is re-renderable (which in general is a good practice for all views)). Also, the expectation is the onboarding view will re-render itself based on any language changes that happen via it's own interface.

I did consider simplifying things and just putting up the dialog on all lang changes, but the problem is that we could theoretically have a client that toggles between servers with different languages and in that case we might have a re-start for a server change, followed by a restart for lang change on every switch.